Src/DependencyCollector/Shared/Implementation/ProfilerSqlProcessingBase.cs

```diff
@@ -253,6 +251,80 @@ private object OnBegin(object thisObj)
         }
 
         /// <summary>
+        ///  Common helper for all EndAsync Callbacks.
+        /// </summary>
+        /// <param name="taskObj">Returned task by the async method.</param>
+        /// <param name="thisObj">This object.</param>
+        private void OnEndAsync(object taskObj, object thisObj)
+        {
+            try
+            {
+                Task task = taskObj as Task;
+                if (task == null)
+                {
+                    DependencyCollectorEventSource.Log.NotExpectedCallback(0, "OnEndAsyncSql", "task == null");
+                    return;
+                }
+
+                DependencyCollectorEventSource.Log.EndAsyncCallbackCalled(thisObj.GetHashCode().ToString(CultureInfo.InvariantCulture));
+
+                task.ContinueWith(t =>
+                {
+                    Exception exceptionObj = null;
+                    if (t.IsFaulted)
+                    {
+                        exceptionObj = t.Exception != null && t.Exception.InnerException != null ? t.Exception.InnerException : t.Exception;
```

Comment:
yes, i thought about - it's definitely improvement, but from the other side we'll keep thisObj object in the table and here i;m removing it in any case even if task is not faulted. let me know what you think about this.

---
In reply to: [98079670](https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/298#discussion_r98079670) [](ancestors = 98079670)